components/counter/index.tsx

```diff
@@ -139,38 +147,42 @@ const CircularProgressBar = (props) => {
 	);
 };
 
-/**
- * Counter
- *
- * @description display character count and limit.
- *
- * @returns <Counter />
- */
-const Counter = forwardRef((props, ref) => {
-	const { count, limit } = props;
-	const percentage = (count / limit) * 100;
-	return (
-		<StyledComponentContext cacheKey="tenup-component-counter">
-			<StyledCounter
-				className={cx('tenup--block-components__character-count', {
-					'is-over-limit': count > limit,
-				})}
-				{...props}
-				ref={ref}
-			>
-				<div className="tenup--block-components__character-count__label">
-					<span className="tenup--block-components__character-count__count">{count}</span>{' '}
-					/{' '}
-					<span className="tenup--block-components__character-count__limit">{limit}</span>
-				</div>
-				<CircularProgressBar percentage={percentage} />
-			</StyledCounter>
-		</StyledComponentContext>
-	);
-});
-
-CircularProgressBar.propTypes = {
-	percentage: PropTypes.number.isRequired,
-};
+interface CounterProps {
+	/**
+	 * Current count.
+	 */
+	count: number;
+
+	/**
+	 * Max limit.
+	 */
+	limit: number;
+}
+
+const Counter: ForwardRefExoticComponent<PropsWithoutRef<CounterProps> & RefAttributes<HTMLDivElement>> = forwardRef<HTMLDivElement, CounterProps>(
+	({
+		count,
+		limit,
+	}, ref ) => {
+		const percentage = (count / limit) * 100;
+		return (
+			<StyledComponentContext cacheKey="tenup-component-counter">
+				<StyledCounter
+					className={cx('tenup--block-components__character-count', {
+						'is-over-limit': count > limit,
+					})}
+					ref={ref}
+				>
```

Comment:
Previously the rest of the props were passed down to the `StlyedCounter` component. But as this doesn't accept any props (as we can see above where it is declared using Emotion styled component), therefore I think, we don't need to pass the rest of the props down to this component.

Reply:
I believe the rest is needed. The `StyledCounter` in the end is only a `styled.div` so a regular DIV which accepts any of the regular HTMLElementDiv attributes. If you for example pass an additional data attribute that should appear on the div.